src/systems/pose_publisher/PosePublisher.cc

```diff
@@ -394,20 +396,15 @@ void PosePublisherPrivate::InitializeEntitiesToPublish(
         (collision && this->publishCollisionPose) ||
         (sensor && this->publishSensorPose);
 
-    // for backward compatibility, top level model pose will be published
-    // if publishNestedModelPose is set to true unless the user explicity
-    // disables this by setting publishModelPose to false
     if (isModel)
     {
       if (parent)
       {
         auto nestedModel = _ecm.Component<components::Model>(parent->Data());
         if (nestedModel)
           fillPose = this->publishNestedModelPose;
-      }
-      if (!fillPose)
-      {
-        fillPose = this->publishNestedModelPose && this->publishModelPose;
+        else
```

Comment:
So, if I understand this correctly, the truth table is:
   | `publish_model_pose` | `publish_nested_model_pose` | Top level model is published |
  |-- | -- | -- | 
  | T | T | T |
  | T | F | T |
  | T | X | F |
  | F | T | F |
  | F | F | F |
  | F | X | F |
  | X | T | T (This is what we want to change in gz-sim10?)|
  | X | F | F |
  | X | X | F |
  


   whereas before, it was

   | `publish_model_pose` | `publish_nested_model_pose` | Top level model is published |
  |-- | -- | -- | 
  | T | T | T |
   | T | F | F (This is the only difference) |
  | T | X | F |
  | F | T | F |
  | F | F | F |
  | F | X | F |
  | X | T | T |
  | X | F | F |
  | X | X | F |

Reply:
yes my interpretation is the same as what's described in the tables above.  

The code change in gz-sim10 would be:

```diff

   this->dataPtr->publishModelPose =
     _sdf->Get<bool>("publish_model_pose",
-        this->dataPtr->publishNestedModelPose).first;
+        this->dataPtr->publishdModelPose).first;
```

and the entry in that table would become:

  | `publish_model_pose` | `publish_nested_model_pose` | Top level model is published |
  |-- | -- | --|
  |X | T | F|